src/aggregator/client/queue.go

```diff
@@ -208,24 +209,63 @@ func (q *queue) Close() error {
 }
 
 func (q *queue) Flush() {
+	var (
+		buf = _queueConnWriteBufPool.Get().(*[]byte)
+		n   int
+		err error
+	)
+
+	for err == nil {
+		// flush everything in batches, to make sure no single payload is too large,
+		// to prevent a) allocs and b) timeouts due to big buffer IO taking too long.
+		var processed int
+		processed, err = q.flush(buf)
+		n += processed
+	}
+
+	if err != nil && err != io.EOF {
+		q.log.Error("error writing data",
+			zap.String("target_instance_id", q.instance.ID()),
+			zap.String("target_instance", q.instance.Endpoint()),
+			zap.Int("bytes_processed", n),
+			zap.Error(err),
+		)
+	}
+
+	// Check buffer capacity, not length, to make sure we're not pooling slices that are too large.
+	// Otherwise, it could result in multi-megabyte slices hanging around, in case we get a single massive write.
+	if cap(*buf) < _queueMaxWriteBufSize {
```

Comment:
Did you mean to check `cap(*buf) > _queueMaxWriteBufSize` here instead of `<`?

Reply:
Oh, it puts the buf back in the pool if it is less than that size. Might use `<=` here.

Reply:
agreed RE `<=`